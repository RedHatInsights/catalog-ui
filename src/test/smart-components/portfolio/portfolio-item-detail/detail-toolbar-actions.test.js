import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';

import DetailToolbarActions from '../../../../smart-components/portfolio/portfolio-item-detail/detail-toolbar-actions';

const prepareTruthyCapability = (truthyCapability) => ({
  copy: false,
  update: false,
  ...(truthyCapability
    ? {
        [truthyCapability]: true
      }
    : {})
});

describe('<DetailToolbarActions />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      orderUrl: 'foo/bar',
      editUrl: 'foo/baz',
      isOpen: false,
      setOpen: jest.fn(),
      copyUrl: 'foo/copy',
      availability: 'available',
      editSurveyUrl: 'foo/edit-survey',
      workflowUrl: 'foo/workflow',
      pathname: 'foo/bar',
      userCapabilities: {
        copy: true,
        update: true
      }
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <DetailToolbarActions {...initialProps} />
      </MemoryRouter>
    );
    expect(toJson(wrapper.find(DetailToolbarActions))).toMatchSnapshot();
  });

  it('should show copy action in dropdown', () => {
    const wrapper = mount(
      <MemoryRouter>
        <DetailToolbarActions
          {...initialProps}
          isOpen
          userCapabilities={prepareTruthyCapability('copy')}
        />
      </MemoryRouter>
    );

    expect(wrapper.find('li')).toHaveLength(3);
    expect(wrapper.find('li#copy-portfolio-item')).toHaveLength(1);
  });

  it('should show edit action in dropdown', () => {
    const wrapper = mount(
      <MemoryRouter>
        <DetailToolbarActions
          {...initialProps}
          isOpen
          userCapabilities={prepareTruthyCapability('update')}
        />
      </MemoryRouter>
    );

    expect(wrapper.find('li')).toHaveLength(3);
    expect(wrapper.find('li#edit-portfolio-item')).toHaveLength(1);
  });
});
