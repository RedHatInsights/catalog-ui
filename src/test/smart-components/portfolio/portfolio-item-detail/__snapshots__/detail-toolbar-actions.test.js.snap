// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailToolbarActions /> should render correctly 1`] = `
<DetailToolbarActions
  availability="available"
  copyUrl="foo/copy"
  editSurveyUrl="foo/edit-survey"
  editUrl="foo/baz"
  isFetching={false}
  isOpen={false}
  orderUrl="foo/bar"
  pathname="foo/bar"
  setOpen={[MockFunction]}
  userCapabilities={
    Object {
      "copy": true,
      "update": true,
    }
  }
  workflowUrl="foo/workflow"
>
  <LevelItem>
    <div>
      <CatalogLink
        isDisabled={false}
        nav={false}
        pathname="foo/bar"
        preserveSearch={true}
        searchParams={Object {}}
      >
        <Styled(Link)
          isDisabled={false}
          to={
            Object {
              "pathname": "foo/bar",
              "search": "",
            }
          }
        >
          <Link
            className="sc-AxiKw drCTeX"
            isDisabled={false}
            to={
              Object {
                "pathname": "foo/bar",
                "search": "",
              }
            }
          >
            <LinkAnchor
              className="sc-AxiKw drCTeX"
              href="foo/bar"
              isDisabled={false}
              navigate={[Function]}
            >
              <a
                className="sc-AxiKw drCTeX"
                href="foo/bar"
                isDisabled={false}
                onClick={[Function]}
              >
                <ButtonWithSpinner
                  id="order-portfolio-item"
                  isDisabled={false}
                  showSpinner={false}
                  variant="primary"
                >
                  <Styled(Component)
                    id="order-portfolio-item"
                    isDisabled={false}
                    variant="primary"
                  >
                    <Component
                      className="sc-AxjAm jhCAUI"
                      id="order-portfolio-item"
                      isDisabled={false}
                      variant="primary"
                    >
                      <ComponentWithOuia
                        component={[Function]}
                        componentProps={
                          Object {
                            "children": Array [
                              <styled.span
                                showSpinner={false}
                              >
                                Order
                              </styled.span>,
                              false,
                            ],
                            "className": "sc-AxjAm jhCAUI",
                            "id": "order-portfolio-item",
                            "isDisabled": false,
                            "variant": "primary",
                          }
                        }
                        consumerContext={null}
                      >
                        <Button
                          className="sc-AxjAm jhCAUI"
                          id="order-portfolio-item"
                          isDisabled={false}
                          ouiaContext={
                            Object {
                              "isOuia": false,
                              "ouiaId": null,
                            }
                          }
                          variant="primary"
                        >
                          <button
                            aria-disabled={null}
                            aria-label={null}
                            className="pf-c-button pf-m-primary sc-AxjAm jhCAUI"
                            disabled={false}
                            id="order-portfolio-item"
                            tabIndex={null}
                            type="button"
                          >
                            <styled.span
                              showSpinner={false}
                            >
                              <span
                                className="sc-AxirZ fnlrkz"
                              >
                                Order
                              </span>
                            </styled.span>
                          </button>
                        </Button>
                      </ComponentWithOuia>
                    </Component>
                  </Styled(Component)>
                </ButtonWithSpinner>
              </a>
            </LinkAnchor>
          </Link>
        </Styled(Link)>
      </CatalogLink>
    </div>
  </LevelItem>
  <LevelItem
    style={
      Object {
        "marginLeft": 16,
      }
    }
  >
    <div
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Dropdown
        dropdownItems={
          Array [
            <DropdownItem
              aria-label="Edit Portfolio"
              component={
                <CatalogLink
                  nav={false}
                  pathname="foo/baz"
                  preserveSearch={true}
                  searchParams={Object {}}
                >
                  Edit
                </CatalogLink>
              }
              id="edit-portfolio-item"
              role="link"
            />,
            <DropdownItem
              aria-label="Copy Portfolio"
              component={
                <CatalogLink
                  nav={false}
                  pathname="foo/copy"
                  preserveSearch={true}
                  searchParams={Object {}}
                >
                  Copy
                </CatalogLink>
              }
              id="copy-portfolio-item"
              role="link"
            />,
            <DropdownItem
              aria-label="Set approval"
              component={
                <CatalogLink
                  nav={false}
                  pathname="foo/workflow"
                  preserveSearch={true}
                  searchParams={Object {}}
                >
                  Set approval
                </CatalogLink>
              }
              role="link"
            />,
            <DropdownItem
              aria-label="Edit survey"
              component={
                <CatalogLink
                  nav={false}
                  pathname="foo/edit-survey"
                  preserveSearch={true}
                  searchParams={Object {}}
                >
                  Edit survey
                </CatalogLink>
              }
              role="link"
            />,
          ]
        }
        isOpen={false}
        isPlain={true}
        onSelect={[Function]}
        onToggle={[MockFunction]}
        position="right"
        toggle={
          <KebabToggle
            id="portfolio-item-actions-toggle"
            onToggle={[Function]}
          />
        }
      >
        <Component
          dropdownItems={
            Array [
              <DropdownItem
                aria-label="Edit Portfolio"
                component={
                  <CatalogLink
                    nav={false}
                    pathname="foo/baz"
                    preserveSearch={true}
                    searchParams={Object {}}
                  >
                    Edit
                  </CatalogLink>
                }
                id="edit-portfolio-item"
                role="link"
              />,
              <DropdownItem
                aria-label="Copy Portfolio"
                component={
                  <CatalogLink
                    nav={false}
                    pathname="foo/copy"
                    preserveSearch={true}
                    searchParams={Object {}}
                  >
                    Copy
                  </CatalogLink>
                }
                id="copy-portfolio-item"
                role="link"
              />,
              <DropdownItem
                aria-label="Set approval"
                component={
                  <CatalogLink
                    nav={false}
                    pathname="foo/workflow"
                    preserveSearch={true}
                    searchParams={Object {}}
                  >
                    Set approval
                  </CatalogLink>
                }
                role="link"
              />,
              <DropdownItem
                aria-label="Edit survey"
                component={
                  <CatalogLink
                    nav={false}
                    pathname="foo/edit-survey"
                    preserveSearch={true}
                    searchParams={Object {}}
                  >
                    Edit survey
                  </CatalogLink>
                }
                role="link"
              />,
            ]
          }
          isOpen={false}
          isPlain={true}
          onToggle={[MockFunction]}
          position="right"
          toggle={
            <KebabToggle
              id="portfolio-item-actions-toggle"
              onToggle={[Function]}
            />
          }
        >
          <ComponentWithOuia
            component={[Function]}
            componentProps={
              Object {
                "dropdownItems": Array [
                  <DropdownItem
                    aria-label="Edit Portfolio"
                    component={
                      <CatalogLink
                        nav={false}
                        pathname="foo/baz"
                        preserveSearch={true}
                        searchParams={Object {}}
                      >
                        Edit
                      </CatalogLink>
                    }
                    id="edit-portfolio-item"
                    role="link"
                  />,
                  <DropdownItem
                    aria-label="Copy Portfolio"
                    component={
                      <CatalogLink
                        nav={false}
                        pathname="foo/copy"
                        preserveSearch={true}
                        searchParams={Object {}}
                      >
                        Copy
                      </CatalogLink>
                    }
                    id="copy-portfolio-item"
                    role="link"
                  />,
                  <DropdownItem
                    aria-label="Set approval"
                    component={
                      <CatalogLink
                        nav={false}
                        pathname="foo/workflow"
                        preserveSearch={true}
                        searchParams={Object {}}
                      >
                        Set approval
                      </CatalogLink>
                    }
                    role="link"
                  />,
                  <DropdownItem
                    aria-label="Edit survey"
                    component={
                      <CatalogLink
                        nav={false}
                        pathname="foo/edit-survey"
                        preserveSearch={true}
                        searchParams={Object {}}
                      >
                        Edit survey
                      </CatalogLink>
                    }
                    role="link"
                  />,
                ],
                "isOpen": false,
                "isPlain": true,
                "onToggle": [MockFunction],
                "position": "right",
                "toggle": <KebabToggle
                  id="portfolio-item-actions-toggle"
                  onToggle={[Function]}
                />,
              }
            }
            consumerContext={null}
          >
            <DropdownWithContext
              autoFocus={true}
              className=""
              direction="down"
              dropdownItems={
                Array [
                  <DropdownItem
                    aria-label="Edit Portfolio"
                    component={
                      <CatalogLink
                        nav={false}
                        pathname="foo/baz"
                        preserveSearch={true}
                        searchParams={Object {}}
                      >
                        Edit
                      </CatalogLink>
                    }
                    id="edit-portfolio-item"
                    role="link"
                  />,
                  <DropdownItem
                    aria-label="Copy Portfolio"
                    component={
                      <CatalogLink
                        nav={false}
                        pathname="foo/copy"
                        preserveSearch={true}
                        searchParams={Object {}}
                      >
                        Copy
                      </CatalogLink>
                    }
                    id="copy-portfolio-item"
                    role="link"
                  />,
                  <DropdownItem
                    aria-label="Set approval"
                    component={
                      <CatalogLink
                        nav={false}
                        pathname="foo/workflow"
                        preserveSearch={true}
                        searchParams={Object {}}
                      >
                        Set approval
                      </CatalogLink>
                    }
                    role="link"
                  />,
                  <DropdownItem
                    aria-label="Edit survey"
                    component={
                      <CatalogLink
                        nav={false}
                        pathname="foo/edit-survey"
                        preserveSearch={true}
                        searchParams={Object {}}
                      >
                        Edit survey
                      </CatalogLink>
                    }
                    role="link"
                  />,
                ]
              }
              isGrouped={false}
              isOpen={false}
              isPlain={true}
              onSelect={[Function]}
              onToggle={[MockFunction]}
              ouiaComponentType="Dropdown"
              ouiaContext={
                Object {
                  "isOuia": false,
                  "ouiaId": null,
                }
              }
              position="right"
              toggle={
                <KebabToggle
                  id="portfolio-item-actions-toggle"
                  onToggle={[Function]}
                />
              }
            >
              <div
                className="pf-c-dropdown pf-m-align-right"
                onToggle={[MockFunction]}
              >
                <KebabToggle
                  ariaHasPopup={true}
                  id="portfolio-item-actions-toggle"
                  isOpen={false}
                  isPlain={true}
                  key=".0"
                  onEnter={[Function]}
                  onToggle={[Function]}
                  parentRef={
                    Object {
                      "current": <div
                        class="pf-c-dropdown pf-m-align-right"
                      >
                        <button
                          aria-expanded="false"
                          aria-haspopup="true"
                          aria-label="Actions"
                          class="pf-c-dropdown__toggle pf-m-plain"
                          id="portfolio-item-actions-toggle"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 192 512"
                            width="1em"
                          >
                            <path
                              d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                              transform=""
                            />
                          </svg>
                        </button>
                      </div>,
                    }
                  }
                >
                  <Toggle
                    aria-label="Actions"
                    ariaHasPopup={true}
                    bubbleEvent={false}
                    className=""
                    id="portfolio-item-actions-toggle"
                    isActive={false}
                    isDisabled={false}
                    isFocused={false}
                    isHovered={false}
                    isOpen={false}
                    isPlain={true}
                    isPrimary={false}
                    isSplitButton={false}
                    onEnter={[Function]}
                    onToggle={[Function]}
                    parentRef={
                      Object {
                        "current": <div
                          class="pf-c-dropdown pf-m-align-right"
                        >
                          <button
                            aria-expanded="false"
                            aria-haspopup="true"
                            aria-label="Actions"
                            class="pf-c-dropdown__toggle pf-m-plain"
                            id="portfolio-item-actions-toggle"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 192 512"
                              width="1em"
                            >
                              <path
                                d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                                transform=""
                              />
                            </svg>
                          </button>
                        </div>,
                      }
                    }
                  >
                    <button
                      aria-expanded={false}
                      aria-haspopup={true}
                      aria-label="Actions"
                      className="pf-c-dropdown__toggle pf-m-plain"
                      disabled={false}
                      id="portfolio-item-actions-toggle"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      <EllipsisVIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 192 512"
                          width="1em"
                        >
                          <path
                            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                            transform=""
                          />
                        </svg>
                      </EllipsisVIcon>
                    </button>
                  </Toggle>
                </KebabToggle>
              </div>
            </DropdownWithContext>
          </ComponentWithOuia>
        </Component>
      </Dropdown>
    </div>
  </LevelItem>
</DetailToolbarActions>
`;
